react/react2/week1/my-app/src/SearchBoxFunction.js

```
@@ -13,16 +13,22 @@ const SearchBoxFunction = () => {
   const getData = async () => {
     if (serchInputValue === "") {
       setFetchResult([]);
+
       setLoading("");
       return;
     }
 
     setLoading("loading...");
 
-     await fetch(`https://api.github.com/search/users?q=${serchInputValue}`)
+    fetch(`https://api.github.com/search/users?q=${serchInputValue}`)
       .then((response) => {
         if (!response.ok ) {
-          throw Error("Could not fetch the data");
+          response.json().then((error) => { 
+
+            setError(error.message)
+   
+          })
+          throw Error(response.message);
         }
```

A small improvement would be to reset the fetch results once we encounter an error.

Also we no longer need to throw an error since we are now using setError here.

```
if (!response.ok) {
  response.json().then((error) => {
    setError(error.message);
    setFetchResult([]);
  });
  return;
}
```

---

why do we need a return here @BjornFridal  ?

---

Good question! When we have an error we often need to "exit" our code. The code is setup to handle a successful reponse from the Github API, so that's why we should exit/return before we get to the `return response.json()`. Otherwise that part will start causing us problems.